sharedbuffers/mapped_struct.py

```diff
@@ -441,7 +446,7 @@ def shared_id(obj):
             return obj | LONG_MASK
         elif isinstance(obj, long):
             # Real longs must make room for the flag bits
-            return (obj & long(0xFFFFFFFFFFFFFFFF)) | ((obj >> 64) << 68) | LONG_MASK
+            return (obj & cython.cast(cython.longlong, 0xFFFFFFFFFFFFFFFF)) | ((obj >> 64) << 68) | LONG_MASK
```

Comment:
longlong can't handle that literal. It needs an ulonglong, but since you're doing `obj &` you really need a python long.